share/functions/__fish_print_packages.fish

```diff
@@ -35,6 +35,17 @@ function __fish_print_packages
         return
     end
 
+    # pkg_info on OpenBSD provides versioning info which we want for
+    # installed packages
+    if begin
+            type -q -f pkg_add
+            and test (uname) = "OpenBSD"
+        end
+	# pkg_info can get blocked if there is another instance of pkg_* running.
+        ls -1 /var/db/pkg
```

Comment:
I'd again replace the `ls` with `set -l files /var/db/pkg/*` and `printf '%s\n' $files`.

Reply:
Is there a "fish" way of removing the path from that output? `%s\n` ends up having `/var/db/pkg/yquake2-7.10` - which will break pkg_* tools.

Reply:
Ah, yes, I understand now.

Instead of `printf`, use `string replace '/var/db/pkg/' '' -- $files`.

(Note that all this breaks on files with newlines, but we don't support that with completions in general and in practice it simply does not matter)